.github/scripts/libff.patch

```diff
@@ -0,0 +1,215 @@
+diff --git a/libff/algebra/fields/bigint.tcc b/libff/algebra/fields/bigint.tcc
+index d78e94f..0b08431 100755
+--- a/libff/algebra/fields/bigint.tcc
++++ b/libff/algebra/fields/bigint.tcc
+@@ -25,39 +25,32 @@ bigint<n>::bigint(const unsigned long x) /// Initalize from a small integer
+ template<mp_size_t n>
+ bigint<n>::bigint(const char* s) /// Initialize from a string containing an integer in decimal notation
+ {
+-    size_t l = strlen(s);
+-    unsigned char* s_copy = new unsigned char[l];
+-
+-    for (size_t i = 0; i < l; ++i)
+-    {
+-        assert(s[i] >= '0' && s[i] <= '9');
+-        s_copy[i] = s[i] - '0';
+-    }
++    mpz_t k;
++    mpz_init(k);
+ 
+-    mp_size_t limbs_written = mpn_set_str(this->data, s_copy, l, 10);
+ #ifndef NDEBUG
+-    assert(limbs_written <= n);
+-#else
+-    UNUSED(limbs_written);
++    assert(mpz_size(k) <= n);
+ #endif
+ 
+-    delete[] s_copy;
++    mpz_set_str(k, s, 10);
++    for (size_t i = 0; i < n; i++)
++    {
++        data[i] = mpz_getlimbn(k, i);
++    }
++    mpz_clear(k);
+ }
+ 
+ template<mp_size_t n>
+ bigint<n>::bigint(const mpz_t r) /// Initialize from MPZ element
+ {
+-    mpz_t k;
+-    mpz_init_set(k, r);
++#ifndef NDEBUG
++    assert(mpz_size(r) <= n);
++#endif
+ 
+-    for (size_t i = 0; i < n; ++i)
++    for (size_t i = 0; i < n; i++)
+     {
+-        data[i] = mpz_get_ui(k);
+-        mpz_fdiv_q_2exp(k, k, GMP_NUMB_BITS);
++        data[i] = mpz_getlimbn(r, i);
+     }
+-
+-    assert(mpz_sgn(k) == 0);
+-    mpz_clear(k);
+ }
+ 
+ template<mp_size_t n>
+@@ -127,7 +120,10 @@ size_t bigint<n>::num_bits() const
+         }
+         else
+         {
+-            return ((i+1) * GMP_NUMB_BITS) - __builtin_clzl(x);
++            if(sizeof(x) == sizeof(unsigned long))
++                return ((i+1) * GMP_NUMB_BITS) - __builtin_clzl(x);
++            else
```

Comment:
I'd probably put an `assert` here to make sure it's `sizeof(unsigned long long)`. But maybe I'm too cautious.